### PR TITLE
docs(plugins) Relate SourceMapPlugins to default devtool config.

### PR DESCRIPTION
**Pull request by kremerd**
In development mode `devtool: eval` [is added automatically](https://github.com/webpack/webpack/blob/master/lib/WebpackOptionsDefaulter.js#L39) to the Webpack config. In particular it must be manually deactivated if one wants to use `SourceMapDevToolPlugin` or `EvalSourceMapDevToolPlugin`.

This PR documents this behavior, both when describing development mode, and - more importantly to me - when describing how to use the two relevant plugins.

### CODE REVIEW

File: src/content/plugins/eval-source-map-dev-tool-plugin.md [hunk 1]
```diff
@@ -35,6 +35,8 @@ The following options are supported:
 
 T> Setting `module` and/or `columns` to `false` will yield less accurate source maps but will also improve compilation performance significantly.
 
+T> When using this plugin in [development mode](/concepts/mode/#mode-development), you must manually set `devTools: false`.
```
Comment (montogeek) [inline src/content/plugins/eval-source-map-dev-tool-plugin.md]: Could you please add a complete example, like:
```js
module.exports = {
  // ...
  devTools: false,
}
```

Comment (montogeek) [reply src/content/plugins/eval-source-map-dev-tool-plugin.md]: Also, it is `devtool`, not `devTools`

**Comment by montogeek** [comment]
@EugeneHlushko What is your comment about?

**Comment by EugeneHlushko** [comment]
You shouldnt be passing devtool false as this pr suggest, it is enough to say `devtool: 'source-map'` to use the mentioned plugin and so on for each corresponding plugin

**Comment by EugeneHlushko** [comment]
Or is this about passing custom configuration object?

**Comment by kremerd** [comment]
@EugeneHlushko The PR is about how to configure `EvalSourceMapDevToolPlugin` and `SourceMapDevToolPlugin` explicitly.

If it's sufficient for you to stick with one of the default `devtool` configurations, then sure you should do that. In that case I wouldn't expect you to look up the plugin configuration, though.

On the other hand if you need to configure one of the plugins manually, then you shouldn't forget to disable the `devtool` option as well. Otherwise two plugins will be loaded and this might screw up your source maps. (This happened to me, and it took me days to figure out.)

File: src/content/plugins/eval-source-map-dev-tool-plugin.md [hunk 2]
```diff
@@ -35,11 +35,26 @@ The following options are supported:
 
 T> Setting `module` and/or `columns` to `false` will yield less accurate source maps but will also improve compilation performance significantly.
 
+T> When using this plugin in [development mode](/concepts/mode/#mode-development), you must manually set `devtool: false`.
```
Comment (EugeneHlushko) [inline src/content/plugins/eval-source-map-dev-tool-plugin.md]: This is misleading, it should say that when you want to use this plugin with custom configuration, it is not possible to do so without setting `devtool: false` in development mode. Using this plugin in development mode through shortcut should work just fine e.g. `devtool: 'eval-source-map'`

File: src/content/plugins/eval-source-map-dev-tool-plugin.md [hunk 3]
```diff
@@ -35,11 +35,26 @@ The following options are supported:
...
 
 ## Examples
 
 The following examples demonstrate some common use cases for this plugin.
 
+### Basic Use Case
+
+To write out `devtool: eval-source-map` using the plugin you can use the following configuration:
```
Comment (EugeneHlushko) [inline src/content/plugins/eval-source-map-dev-tool-plugin.md]: Can you rephrase this? I would expect this to note that in development it is necessary to set `devtool: false` in order for custom configured plugin instance to work, default use case (`devtool: 'eval-source-map'`) as mentioned above should work just fine for both development and production modes.
